_posts/2024-08-14-adding-to-bioconda-quickguide.md

```diff
@@ -0,0 +1,468 @@
+
+      ```bash
+      mkdir recipes/<toolname>
+      ```
+
+      The name of the software must formatted in all lower case, and with only letters, numbers, and hyphens.
+
+      If our package is an R package, we should prefix the name with `r-`.
+
+      ⚠ Make sure a tool with the same name doesn't exist!
+      If it does - consider adding a suffix.
+      For example, [`-mg` to indicate software for metagenomics](https://github.com/bioconda/bioconda-recipes/blob/master/recipes/metawrap-mg/meta.yaml), or [`-lite` for a version of a recipe that doesn't include preinstalled databases](https://github.com/bioconda/bioconda-recipes/blob/master/recipes/antismash-lite/meta.yaml).
+
+      Then, create an empty text file called `meta.yaml` in the new directory.
+
+      ```bash
+      touch recipes/<toolname>/meta.yaml
+      ```
+
+3. Add the following sections in the `meta.yaml` file (or double check if already made with `grayskull`).
+   When in doubt, copy from other similar existing recipes already on Bioconda:
+
+   - `package:`
+     - Specify the name (same specifications as above) and version of the tool/package.
+   - `source:`
+     - Specify the URL to the source code tarball or zip file for conda to download.
+     - The e.g. `sha265` hash string of the file for download verification.
+   - `build:`
+     - Specify the build number (for new packages or new software version, always `0`).
+     - Possibly the architecture (e.g. `noarch` for Python packages).
+     - A `run_exports` subpackage pinning.
+   - `requirements:`
+     - Specify a list of the various dependencies of the software needs during various sections of the build process, i.e., `host`, `build`, and `run`.
+     - Should have a minimum versions, and ideally a with [`>=` notation](https://docs.conda.io/projects/conda-build/en/latest/resources/package-spec.html#id3).
+   - `test:`
+     - One or more (e.g. if multiple CLI tools or scripts exist under the package) commands to test the software installed correctly.
+     - Typically simply running the tool with `--help` or `--version` is sufficient, but must have a `0` exit code to indicate success.
+     - If `--help` ends with a non-`0` code, we can try `grep`ing for a string in the help message.
+   - `about:`
+     - URL of such as source code repository or documentation home page.
+     - License type (possibly from a fixed list?) [^4].
+     - Corresponding license file name as in the tarball.
+     - A short one-sentence summary and/or long-form description of the software.
+   - `extras:`
+     - other metadata information such as the DOI identifier of any associated publication the software may have.
+     - Other identifiers of the software.
+
+   ![A relatively simple example conda recipe example for Centrifuge, based on the descriptions above]({% link assets/images/2024-08-14-bioconda-guide/bioconda-guide-centrifugemeta.png %})
```

Comment:
It would be better to use a code block than an image.

Reply:
Agreed, unfortunately, while I did that originally, the double brackets used throughout messed with some stuff with the blog's jekyll stuff.

I couldn't get it the double brackets to escape and not show the escape sequence :(

If you have any suggestions I can try again.

Reply:
darn, maybe @hexylena has an idea?

Reply:
Your {% raw %} trick worked! Can be considered resolved :) Images removed now